pkg/sentry/fsimpl/fuse/connection.go

```diff
@@ -54,24 +76,258 @@ type FutureResponse struct {
 // Connection is the struct by which the sentry communicates with the FUSE server daemon.
 type Connection struct {
 	fd *DeviceFD
+
+	// MaxRead size in bytes
+	MaxRead uint32
+
+	// MaxWrite size in bytes
+	MaxWrite uint32
+
+	// MaxPages is the maximum number of pages for a single request to use
+	MaxPages uint16
+
+	// MaxBackground is the maximum number of outstanding background requests
+	MaxBackground uint16
+
+	// CongestionThreshold for number of background requests
+	CongestionThreshold uint16
+
+	// NumberBackground requests currently
+	NumBackground uint16
+
+	// ActiveBackground requests number currently queued for userspace
+	ActiveBackground uint16
+
+	// TODO: BgQuque
+	// some queue for background queued requests
+
+	// BgLock protects:
+	// MaxBackground, CongestionThreshold, NumBackground,
+	// ActiveBackground, BgQueue, Blocked
+	BgLock sync.Mutex
+
+	// Initialized if INIT reply has been received.
+	// Until it's set, FUSE request allocation will be suspended
+	Initialized bool
+
+	// protects Initialized
+	initializedLock sync.Mutex
+
+	// Blocked when:
+	// before the INIT reply is received,
+	// and if there are too many outstading backgrounds requests
+	Blocked bool
```

Comment:
We may need to add a channel for requests to wait on before the INIT is complete. On completion, init should wake all of these up by closing the channel. this should probably happen in `Connection.callFuture`.

Reply:
Added a TODO.